src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -131,20 +134,51 @@ def invoke_apt_cache(self, command):
     # region Classification-based (incl. All) update check
     def get_all_updates(self, cached=False):
         """Get all missing updates"""
+        all_updates = []
+        all_updates_versions = []
+        ubuntu_pro_client_all_updates_query_success = False
+
         self.composite_logger.log_debug("\nDiscovering all packages...")
-        if cached and not len(self.all_updates_cached) == 0:
-            self.composite_logger.log_debug(" - Returning cached package data.")
-            return self.all_updates_cached, self.all_update_versions_cached  # allows for high performance reuse in areas of the code explicitly aware of the cache
+        # use Ubuntu Pro Client cached list when the conditions are met.
+        if self.__pro_client_prereq_met and not len(self.ubuntu_pro_client_all_updates_cached) == 0:
+            all_updates = self.ubuntu_pro_client_all_updates_cached
+            all_updates_versions = self.ubuntu_pro_client_all_updates_versions_cached
+
+        elif not self.__pro_client_prereq_met and not len(self.all_updates_cached) == 0:
+            all_updates = self.all_updates_cached
+            all_updates_versions = self.all_update_versions_cached
+
+
```

Comment:
Unnecessary extra new line

Reply:
Done

Reply:
@SathishMSFT the extra new line still shows up